### PR TITLE
Release/v1.2.0

### PR DESCRIPTION
**Pull request by matt-sparq**
This update introduces the concept of 'Houston services', which are microservices that can execute a Houston stage regardless of the Houston plan that the stage is a part of. 

New features:
- Compatibility with the open source version of Houston 
  - Self-hosted Houston APIs can be used by setting the HOUSTON_BASE_URL environment variable 
  - Client now looks in the 'services' block to determine how to trigger a stage
- Added houston.gcp.cloud_function.service function decorator
- Add Houston 'commands' and command line interface
- Generalised `Houston.trigger()` method allows for all Houston services to trigger any stage

The client remains backwards compatible with the existing Houston API and plans that don't contain a 'services' attribute.

Other:
- Removed Python 2.7 support
- Refactor to use pyproject.toml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,202 +1,33 @@
...
-
-# linux
-export https_proxy=10.0.0.1
-
+pip install "houston-client[gcp]"
 ```
 
 ## Usage
 
-Please read the documentation before getting started: https://callhouston.io/docs
```
Comment (jawatDSQ) [inline README.md:35]: I feel the README.md would benefit from outlining the functionality areas in this client (used in Python and as a Cloud Function wrapper)

File: tests/test_gcp.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ class TestCallStageViaPubSub(unittest.TestCase):
```
Comment (jawatDSQ) [inline tests/test_gcp.py:45]: Given the amount of increase in service code base - do we need to implement more tests around it?

Comment (matt-sparq) [reply tests/test_gcp.py:45]: I've added a test case for the service wrapper below

File: README.md (hunk 1 above)
Comment (matt-sparq) [reply README.md:35]: I've added an overview to the README and linked to various new documentation articles

**Review by jawatDSQ — APPROVED** [review]

**Pull request merged**